test_plans/oneapi_local_memory.asciidoc

```diff
@@ -0,0 +1,103 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl extension oneapi local memory
+
+This is a test plan for the extension enabling the declaration of local memory
+objects at the kernel functor scope described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc[sycl_ext_oneapi_local_memory.asciidoc]
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_LOCAL_MEMORY` so they can
+be skipped if feature is not supported.
+
+=== Type coverage
+
+Let `T` is the type of an object to be allocated within local memory space.
+
+Tests are performed using the following types as `T`:
+
+In regular mode:
+
+* `int`
+* `float`
+* custom trivially destructible type
```

Comment:
If you plan to use this custom type with `group_local_memory_for_overwrite` then the custom type must have a default constructor.

It would also be nice to test the case when the constructor takes more than one argument, so I think the custom type should also have a constructor that takes several arguments.  This will provide better testing for the `args` parameter pack to `group_local_memory`.  Alternatively, you could have several custom types, one that has a default constructor, and another whose constructor takes several arguments.